DeleteUserDataActivity/backupAndDelete.ts

```diff
@@ -112,21 +108,25 @@ const backupAndDeleteProfile = ({
   readonly userDataBackup: IBlobServiceInfo;
   readonly servicePreferencesModel: ServicePreferencesDeletableModel;
   readonly fiscalCode: FiscalCode;
-}) =>
-  executeRecursiveBackupAndDelete<RetrievedProfile>(
-    item => profileModel.deleteProfileVersion(item.fiscalCode, item.id),
-    userDataBackup,
-    item => `profile/${item.id}.json`,
-    profileModel.findAllVersionsByModelId(fiscalCode)
-  ).chain(_ =>
-    executeRecursiveBackupAndDelete<RetrievedServicePreference>(
-      item => servicePreferencesModel.delete(item.id, item.fiscalCode),
+}): TE.TaskEither<DataFailure, true> =>
+  pipe(
+    executeRecursiveBackupAndDelete<RetrievedProfile>(
+      item => profileModel.deleteProfileVersion(item.fiscalCode, item.id),
       userDataBackup,
-      item => `service-settings/${item.id}.json`,
-      servicePreferencesModel.findAllByFiscalCode(fiscalCode)
-    ).foldTaskEither<DataFailure, ReadonlyArray<RetrievedServicePreference>>(
-      _df => te.taskEither.of([]),
-      te.taskEither.of
+      item => `profile/${item.id}.json`,
+      profileModel.findAllVersionsByModelId(fiscalCode)
+    ),
+    TE.chainW(_ =>
+      executeRecursiveBackupAndDelete<RetrievedServicePreference>(
+        item => servicePreferencesModel.delete(item.id, item.fiscalCode),
+        userDataBackup,
+        item => `service-settings/${item.id}.json`,
+        servicePreferencesModel.findAllByFiscalCode(fiscalCode)
+      )
+    ),
+    TE.foldW(
+      () => TE.of(true as const),
```

Comment:
Do we need `as const` ?

Reply:
We do, otherwise true would be considered as a boolean